src/main/java/org/jenkinsci/plugins/workflow/support/storage/SimpleXStreamFlowNodeStorage.java

```diff
@@ -71,11 +69,11 @@
 public class SimpleXStreamFlowNodeStorage extends FlowNodeStorage {
     private final File dir;
     private final FlowExecution exec;
-    private final LoadingCache<String,FlowNode> nodeCache = CacheBuilder.newBuilder().softValues().build(new CacheLoader<String,FlowNode>() {
-        @Override public FlowNode load(String key) throws Exception {
-            return SimpleXStreamFlowNodeStorage.this.load(key).node;
-        }
-    });
+
+    private final LoadingCache<String,FlowNode> nodeCache = Caffeine.newBuilder()
+            .weakKeys()
```

Comment:
Also, why weak _keys_? These should be small `String`s, currently always numeric like `123`. We could perhaps `intern` them, but beyond that…?